xarray/tests/test_plot.py

```diff
@@ -2391,11 +2428,12 @@ def test_plot_transposed_nondim_coord(plotfunc):
 def test_plot_transposes_properly(plotfunc):
     # test that we aren't mistakenly transposing when the 2 dimensions have equal sizes.
     da = xr.DataArray([np.sin(2 * np.pi / 10 * np.arange(10))] * 10, dims=("y", "x"))
-    hdl = getattr(da.plot, plotfunc)(x="x", y="y")
-    # get_array doesn't work for contour, contourf. It returns the colormap intervals.
-    # pcolormesh returns 1D array but imshow returns a 2D array so it is necessary
-    # to ravel() on the LHS
-    assert np.all(hdl.get_array().ravel() == da.to_masked_array().ravel())
+    with figure_context():
+        hdl = getattr(da.plot, plotfunc)(x="x", y="y")
+        # get_array doesn't work for contour, contourf. It returns the colormap intervals.
+        # pcolormesh returns 1D array but imshow returns a 2D array so it is necessary
+        # to ravel() on the LHS
+        assert np.all(hdl.get_array().ravel() == da.to_masked_array().ravel())
```

Comment:
thanks for updating. Looking through the other tests there are a few other calls to `assert np.all(... == ...)`. It would be good to switch those to `numpy.testing.assert_equal` or `xarray.testing.assert_duckarray_equal`, too, but that doesn't have to be done in this PR.